lib/actions/appleMusic.js

```diff
@@ -13,46 +13,172 @@ function getAlbumUri(id) {
+              Promise.reject('No tracks were streamable');
+            } else {
+              var artistCount = 1;
+              var trackCount = 1;
+              var artists = tracksJson.results.map(function(track) {
+                return track.artistName.toLowerCase();
+              }).sort();
+              var tracks = tracksJson.results.map(function(track) {
+                return track.trackName.toLowerCase();
+              }).sort();
+
+              // Determine if the search results are many tracks for an artist OR a specific song title.
+              var prevArtist=artists[0];
+              var prevTrack=tracks[0];
+  
+              for (var i=1; i < tracksJson.resultCount;i++) {
+                if (artists[i] != prevArtist) {
+                  artistCount++;
+                  prevArtist = artists[i];
+                }
+                if (tracks[i] != prevTrack) {
+                  trackCount++;
+                  prevTrack = tracks[i];
+                }  
+              }
+          
+              // If there are at least twice as many track differences that artist differences, then load all of the artist's tracks into the queue, ELSE play just the first track now (iTunes returns the most popular track first) 
+              if (trackCount/artistCount > 2) {
+                const queueURI = 'x-rincon-queue:' + player.uuid + '#0';
+                var first = true;
+
+                return player.coordinator.removeAllTracksFromQueue()
+                  .then(() => player.setAVTransportURI(queueURI, ''))
+                  .then(() => {
+                    var trackPlayList = tracksJson.reduce(function(newArray, track){
+                      if (track.isStreamable) {
+                        var skip = false;
+  
+                        for (var j=0; (j < newArray.length) && !skip ; j++) {
+                          skip = (track.trackName == newArray[j]);
+                        }
+                    
+                        if (!skip) {
+                          const metadataID = METADATA_URI_STARTERS[type] + encodeURIComponent(track.trackID);
+                          const metadata = getMetadata(metadataID, PARENTS[type], CLASSES[type]);
+                          const uri = uriTemplates[type](encodeURIComponent(track.trackID));
+                          const nextTrackNo = player.coordinator.state.trackNo + 1;
+
+                          newArray.push(track.trackName);
+                          player.coordinator.addURIToQueue(uri, metadata, true, nextTrackNo) 
+                            .then(() => {
+                              if (first) {
+                                first = false;
+                                player.coordinator.play();
+                              }
+                            });
+                        }
+                      } 
+                      return newArray; /* This is important! */
+                    }, []);
+                  });
```

Comment:
This wouldn't work, it would still fire away requests as soon as it is finished. 

```
return tracksJson.reduce(function (promise, track) {
  // prepare stuff
  return promise.then(() => player.coordinator..addURIToQueue(uri, metadata, true, nextTrackNo));
}, Promise.resolve());
```

This would be the way to execute it in series, which basically does:
```
Promise.resolve().then(() => player.addURIToQueue()).then(() => player.addURIToQueue()).then(() => player.addURIToQueue()).then(() => player.addURIToQueue()).then(() => player.addURIToQueue()).then(() => player.addURIToQueue()) ... and on
```

You could either wait for this to finish and then play, or do a similar approach and hit play after first track has been added like you do here.